docs/user-guide/extending_argocd_with_plugins.md

```diff
@@ -0,0 +1,132 @@
+# Extend argocd with plugins
+
+## Overview
+
+This guide demonstrates how to write plugins for the
+`argocd` CLI tool. Plugins are a mean to extend `argocd` CLI with new sub-commands,
+allowing for new and custom features which are not included in the main distribution
+of `argocd` CLI.
+
+## Prerequisites
+
+You need to have a working `argocd` binary installed locally. You can follow
+the [cli installation documentation](https://argo-cd.readthedocs.io/en/stable/cli_installation/) to install the binary.
+
+## Create `argocd` plugins
+
+A plugin is a standalone executable file, whose name begins with argocd-.
+To install a plugin, move its executable file to anywhere on your `PATH`.
+
+`argocd` allows plugins to add custom commands such as `argocd my-plugin bla bla` by providing a 
```

Comment:
better to rephrase here and not provide informal examples in the official documentation. So, instead of:
`argocd my-plugin bla bla`
I suggest to change to something like:
`argocd my-plugin arg1 arg2 --flag1`